pkg/ebpf/net_proto.go

```diff
@@ -14,16 +14,41 @@ import (
 type protocolHandler func(*bufferdecoder.EbpfDecoder, *trace.Event) error
 
 // protocolProcessor calls handlers of the appropriate protocol event
-func protocolProcessor(networkThread procinfo.ProcessCtx, evtMeta bufferdecoder.NetEventMetadata, decoder *bufferdecoder.EbpfDecoder) (trace.Event, error) {
+func protocolProcessor(networkThread procinfo.ProcessCtx, evtMeta bufferdecoder.NetEventMetadata, decoder *bufferdecoder.EbpfDecoder, ifaceName string) (trace.Event, error) {
 
 	// create network event without any args
 	evt := CreateNetEvent(evtMeta, networkThread, EventsDefinitions[evtMeta.NetEventId].Name)
 
 	// handle specific protocol data
 	err := callProtocolHandler(evtMeta.NetEventId, decoder, &evt)
+	if err != nil {
+		return evt, err
+	}
+
+	// resolve interface name in 'metadata' arg
+	err = resolveIfaceName(&evt, "metadata", ifaceName)
+
 	return evt, err
 }
 
+// resolveIfaceName replaces the interface number with interface name for 'metadata' arg
+func resolveIfaceName(evt *trace.Event, argName string, ifaceName string) error {
+	for idx, arg := range evt.Args {
+		if arg.Name == argName {
+			if metadata, isPktMeta := arg.Value.(trace.PktMeta); isPktMeta {
+				metadata.Iface = ifaceName
+				evt.Args[idx] = trace.Argument{
+					ArgMeta: arg.ArgMeta,
```

Comment:
Don't we need to change the argument metadata in this case as well? (As the type is changed from int to string)